### PR TITLE
Creating maximal and minimal lenght for providers

### PR DESCRIPTION
**Pull request by Kalbra**
This PR works on #1506 by adding maximal and minimal lenght for providers.

### TODO:
Method changing:
- [ ] Adress
- [ ] Company
- [ ] Job
- [ ] Lorem
- [ ] Person

### CODE REVIEW

File: faker/providers/__init__.py [hunk 1]
```diff
@@ -464,6 +469,24 @@ def random_elements(
                        ("d", 0.05),
                    ]), unique=True
         """
+        # Check for max and min
+        if max_element_length or min_element_length:
+            appropriate_elements = []
+            if not max_element_length:
+                for element in elements:
+                    if len(element) >= min_element_length:
```
[the remark this answers is not in the document]
Comment (Kalbra) [reply faker/providers/__init__.py]: Well it somehow works, but I get what you mean. I changed that in [28087b4](https://github.com/joke2k/faker/pull/1615/commits/28087b4a77f449b616d4c10d430417734d093e18) by using a more logic way.

File: faker/providers/person/__init__.py [hunk 2]
```diff
@@ -235,92 +235,94 @@ def name_female(self) -> str:
...
+    def suffix(self, min_length: Optional[int] = None, max_length: Optional[int] = None) -> str:
         if hasattr(self, "suffixes"):
-            return self.random_element(self.suffixes)  # type: ignore[attr-defined]
+            return self.random_element(self.suffixes, min_length, max_length)  # type: ignore[attr-defined]
         if hasattr(self, "suffixes_male") and hasattr(self, "suffixes_female") and hasattr(self, "suffixes_nonbinary"):
-            suffixes: Sequence[ElementsType] = self.random_element(
-                (self.suffixes_male, self.suffixes_female, self.suffixes_nonbinary)  # type: ignore[attr-defined]
-            )
-            return self.random_element(suffixes)
+            suffixes: Sequence[ElementsType] = self.suffixes_male
```
[the remark this answers is not in the document]
Comment (Kalbra) [reply faker/providers/person/__init__.py]: Even if I do not know exactly what `# type: ignore[attr-defined]` do I added it at [d7dc9d9](https://github.com/joke2k/faker/pull/1615/commits/d7dc9d90d8c3b458587f73358c4595e85fde1c91) to prefix and suffix.

**Review by fcurella — CHANGES_REQUESTED** [review]

**Comment by Kalbra** [comment]
I was just thinking about using args and kwargs instead of `min_length: Optional[int] = None, max_length: Optional[int] = None`
because then it would be easier to add parameters for future features.
@fcurella what do you think?

File: faker/providers/person/__init__.py [hunk 3]
```diff
@@ -235,92 +235,94 @@ def name_female(self) -> str:
...
         if hasattr(self, "prefixes"):
-            return self.random_element(self.prefixes)  # type: ignore[attr-defined]
+            return self.random_element(self.prefixes, min_length, max_length)  # type: ignore[attr-defined]
         if hasattr(self, "prefixes_male") and hasattr(self, "prefixes_female") and hasattr(self, "prefixes_nonbinary"):
-            prefixes: Sequence[ElementsType] = self.random_element(
-                (self.prefixes_male, self.prefixes_female, self.prefixes_nonbinary)  # type: ignore[attr-defined]
-            )
-            return self.random_element(prefixes)
+            prefixes: Sequence[ElementsType] = self.prefixes_male  # type: ignore[attr-defined]
+            prefixes.update(self.prefixes_female)
```
[the remark this answers is not in the document]
Comment (Kalbra) [reply faker/providers/person/__init__.py]: I added this in [53f2530](https://github.com/joke2k/faker/pull/1615/commits/53f2530c504b85377766d32f3dec159d94620532)

**Comment by fcurella** [comment]
@Kalbra 
> I  was just thinking about using args and kwargs

Wouldn't that break automcompletion with mypy? I'd rather keep the arguments explicit.

**Comment by Kalbra** [comment]
@fcurella 
> Wouldn't that break automcompletion with mypy? I'd rather keep the arguments explicit.

Good point, makes sense to me, so I will keep the arguments explicit.

**Comment by MetcalfeTom** [comment]
Is this PR still in active development?  Would be a very useful feature for our team.

If it is abandoned, I'd be happy to contribute myself

**Comment by fcurella** [comment]
Closed in favor of https://github.com/joke2k/faker/pull/1662

**Pull request closed without merging**